### PR TITLE
Remove momentum updating from val step and add separate val queue

### PR DESCRIPTION
**Pull request by maximzubkov**
## What does this PR do?

This pull request fixes the problem discussed in the following issues https://github.com/PyTorchLightning/lightning-bolts/issues/615 https://github.com/PyTorchLightning/lightning-bolts/issues/630 

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## Changes:
- Remove `_momentum_update_key_encoder` and `_dequeue_and_enqueue` from `forward`
- Add separate `val_queue` and `val_queue_ptr`
- Return `key` form `forward`

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: pl_bolts/models/self_supervised/moco/moco2_module.py [hunk 1]
```diff
@@ -121,6 +121,12 @@ def __init__(
 
         self.register_buffer("queue_ptr", torch.zeros(1, dtype=torch.long))
 
+        # create the validation queue
+        self.register_buffer("val_queue", torch.randn(emb_dim, num_negatives))
+        self.queue = nn.functional.normalize(self.val_queue, dim=0)
```
Comment (maximzubkov) [inline pl_bolts/models/self_supervised/moco/moco2_module.py:126]: As far as I understood, there is a small bug here, we should assign normalized `self.val_queue` to itself, but not to `self.queue`. I will fix it in the following commits

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@ananyahjha93 mind check

**Review by akihironitta — APPROVED** [review]
@maximzubkov Thank you for sending the fix! LGTM!

@ananyahjha93 Could you double-check the change here?
